### DISCUSSION TITLE
why can't I use <this> keyword for calling an instance method in this below code

### DISCUSSION THREAD

**Discussion by shura-zerotwo** [community/community#28454] [Codespaces]
public class thisDemo {
	
	int a = 100;
	static int b = 200;
	
	void demo1() {
		int a = 10;
		int b = 20;
		System.out.println("a="+a);
		System.out.println("instance variable a="+this.a);
		System.out.println("b="+b);
		System.out.println("static variable b="+thisDemo.b);
	} //instance method()
	
	static void demo2() {
		int c = 30;
		int d = 40;
		System.out.println("I am a static method, i can be called directly by the class name");
	} //static method()
	
	public static void main(String[] args) {
		
		this.demo1();
		
		demo2();
	
	}
	
}

**Comment by airtower-luna** [comment]
`this` refers to the current object, so you can only use it in methods of an object. `main()` is a `static` function, static means it's *not* tied to any object. And because there's no current object, you can't use `this` and non-static methods.

You first need to create an object (an *instance* of the class) using `new` and call the method on that object.

For example, you can create an empty `String` object like this, and then ask it for its length by calling [its `length()` method](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#length()):

```java
String s = new String();
System.out.println(s.length());
```

This will print `0`, because the string is empty. The point is you need to ask for the length of *that specific* String (the object/instance). You can't ask for the length of a String in general.
